src/main/java/net/nicoulaj/maven/plugins/checksum/mojo/ArtifactsMojo.java

```diff
@@ -119,9 +122,29 @@
      * @see #individualFiles
      * @since 1.4
      */
-    @Parameter (defaultValue = "false")
+    @Parameter( defaultValue = "false" )
     protected boolean appendFilename;
 
+    /**
+     * Whether to skip checksum generation for the project's main artifact.
+     */
+    @Parameter( defaultValue = "false" )
+    protected boolean excludeMainArtifact;
+
+    /**
+     * Comma-separated list of classifiers. For all secondary project artifacts having any of the given classifiers checksums are generated.
+     * If not set all secondary project artifacts are considered.
+     */
+    @Parameter
+    protected String includeClassifiers;
```

Comment:
Maybe it would be better to use the `List` type here ? `includes`/`excludes` usually use children elements rather than a single string + parsing.

Reply:
I use the constructor of `ClassifierFilter` which only understands comma-separated strings (https://maven.apache.org/shared/maven-common-artifact-filters/apidocs/org/apache/maven/shared/artifact/filter/collection/ClassifierFilter.html#ClassifierFilter-java.lang.String-java.lang.String-), but this is rather an implementation detail, so I am fine with converting to a List as well.

Reply:
The problem is that converting the list back to a comma separated string requires quite some code or external dependencies with Java < 8. Is it ok to require Java 8 for the upcoming release?

Reply:
Yes, it's OK, but nevermind, I did not think this would require additional code. I am fine with this implementation.